components/src/main/java/org/wso2/carbon/messaging/StreamingCarbonMessage.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.messaging;
+
+import java.io.InputStream;
+
+/**
+ * CarbonMessage which can transport an input stream.
```

Comment:
Please review the improved comment in https://github.com/wso2/carbon-messaging/pull/50/commits/99d2709f3c2ae2be20abc576451632d9a91e3d15